systemtest/src/test/java/io/strimzi/systemtest/report/DefaultClusterReportST.java

```diff
@@ -0,0 +1,445 @@
+            clusterName + "-entity-topic-operator-config.yaml",
+            clusterName + "-entity-user-operator-config.yaml",
+            clusterName + "-kafka-0.yaml",
+            clusterName + "-kafka-1.yaml",
+            clusterName + "-kafka-2.yaml",
+            clusterName + "-zookeeper-config.yaml",
+            "my-connect-connect-config.yaml",
+            "my-mm2-mirrormaker2-config.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/configmaps", ConfigMap.class, s, 1);
+        }
+    }
+
+    private void assertValidCustomResourceDefinitions(String outPath) throws IOException {
+        for (String s : Arrays.asList(
+            "kafkabridges.kafka.strimzi.io.yaml",
+            "kafkaconnects.kafka.strimzi.io.yaml",
+            "kafkamirrormaker2s.kafka.strimzi.io.yaml",
+            "kafkarebalances.kafka.strimzi.io.yaml",
+            "kafkas.kafka.strimzi.io.yaml",
+            "kafkatopics.kafka.strimzi.io.yaml",
+            "kafkausers.kafka.strimzi.io.yaml",
+            "strimzipodsets.core.strimzi.io.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/customresourcedefinitions", CustomResourceDefinition.class, s, 1);
+        }
+    }
+
+    private void assertValidCustomDeployments(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            "my-bridge-bridge.yaml",
+            clusterName + "-cruise-control.yaml",
+            clusterName + "-entity-operator.yaml",
+            "strimzi-cluster-operator.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/deployments", Deployment.class, s, 1);
+        }
+    }
+
+    private void assertValidNetworkPolicies(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            clusterName + "-entity-operator.yaml",
+            clusterName + "-network-policy-cruise-control.yaml",
+            clusterName + "-network-policy-kafka.yaml",
+            clusterName + "-network-policy-zookeeper.yaml",
+            "my-mm2-mirrormaker2.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/networkpolicies", NetworkPolicy.class, s, 1);
+        }
+    }
+
+    private void assertValidPodDisruptionBudgets(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            "my-bridge-bridge.yaml",
+            clusterName + "-kafka.yaml",
+            clusterName + "-zookeeper.yaml",
+            "my-connect-connect.yaml",
+            "my-mm2-mirrormaker2.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/poddisruptionbudgets", PodDisruptionBudget.class, s, 1);
+        }
+    }
+
+    private void assertValidPods(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            clusterName + "-kafka-0.yaml",
+            clusterName + "-kafka-1.yaml",
+            clusterName + "-kafka-2.yaml",
```

Comment:
I would use methods from API instead of "hardcoding" it here.
So f.e. `KafkaResources.kafkaPodName()` etc.

This applies for all other things -> `KafkaMirrorMaker2Resources.deploymentName(..)`

Also, we have some helper methods for Kafka with/without NodePools, if you want to use them (in `KafkaResource` class).

Reply:
I'm taking these values from testStorage. Can you be more specific?

Reply:
I mean for the file names use methods from API -> for example instead of `clusterName + "-kafka-0.yaml",` use `KafkaResources.kafkaPodName(clusterName, 0) + ".yaml"`

So instead of having those values hardcoded, use methods from API module, so in case of any change, we will not have to fix the test.

Reply:
The resources are spread across `*Resources` classes, but regarding nodepools, I guess you will have to use the ones in `KafkaResource` class in `systemtest` module

Reply:
I'm confused now. Before you were mentioning `KafkaResources` in api module, now `KafkaResource` and other `*Resource` classes in systemtest module. 

For example, I can get the first kafka pod name with both `KafkaResources.kafkaPodName(clusterName, 0)` and `KafkaResource.getKafkaPodName("my-cluster", 0)`.

Which classes do I have to use?

Reply:
In case you have testcase, where you either use NodePools mode or non-NodePools mode based on the env variable, you should use `KafkaResource.getKafkaPodName()` in the `systemtest` module. In other case, you can use `KafkaResources` (and other classes) in `api` module